### ISSUE TITLE
[5.x]: Calling unknown method: craft\models\Site::getStore()

### ISSUE THREAD

**Issue by boboldehampsink** [craftcms/commerce#3514]
### Description

Seeing this when trying to start my site with Commerce 5:

```
web-1  | 2024-05-22 09:56:11 [console.ERROR] [yii\base\UnknownMethodException] yii\base\UnknownMethodException: Calling unknown method: craft\models\Site::getStore() in /app/user/vendor/yiisoft/yii2/base/Component.php:300
web-1  | Stack trace:
web-1  | #0 /app/user/vendor/craftcms/commerce/src/web/twig/Extension.php(61): yii\base\Component->__call('getStore', Array)
web-1  | #1 /app/user/vendor/twig/twig/src/ExtensionSet.php(331): craft\commerce\web\twig\Extension->getGlobals()
web-1  | #2 /app/user/vendor/twig/twig/src/Environment.php(787): Twig\ExtensionSet->getGlobals()
web-1  | #3 /app/user/vendor/twig/twig/src/Environment.php(800): Twig\Environment->getGlobals()
web-1  | #4 /app/user/vendor/twig/twig/src/Template.php(367): Twig\Environment->mergeGlobals(Array)
web-1  | #5 /app/user/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
web-1  | #6 /app/user/vendor/twig/twig/src/TemplateWrapper.php(38): Twig\Template->render(Array)
web-1  | #7 /app/user/vendor/craftcms/cms/src/web/View.php(665): Twig\TemplateWrapper->render(Array)
```

Probably because it isn't registered yet when starting Craft. It should be null-checked.

### Steps to reproduce

1. Seeing this with Commerce + SEOmatic installed

### Expected behavior

No error

### Actual behavior

Error

### Craft CMS version

5

### Craft Commerce version

5.1.6

### PHP version

8.3

### Database type and version

PostgreSQL

### Installed plugins and versions

-

**Comment by nfourtythree** [comment]
Hi @boboldehampsink 

Thank you for your message, unfortunately, we are unable to replicate what you are seeing.

Could you please send your project files (`composer.json`, `composer.lock`, database backup and project config directory) along with specific steps to replicate the error to support@craftcms.com. Hopefully, we can see the issue first-hand and get things sorted ASAP.

Thanks!

**Comment by lukeholder** [comment]
Related: https://github.com/craftcms/commerce/issues/3498

**Comment by boboldehampsink** [comment]
It's probably got to do with the twig extension being registered before the event listeners

**Comment by boboldehampsink** [comment]
Fixed in 5.0.7
